clippy_utils/src/qualify_min_const_fn.rs

```diff
@@ -306,7 +328,12 @@ fn check_terminator<'tcx>(
         | TerminatorKind::Terminate
         | TerminatorKind::Unreachable => Ok(()),
 
-        TerminatorKind::Drop { place, .. } => check_place(tcx, *place, span, body),
+        TerminatorKind::Drop { place, .. } => {
+            for local in &body.local_decls {
+                check_ty(tcx, local.ty, span, true)?;
+            }
+            check_place(tcx, *place, span, body)
```

Comment:
This shouldn't be necessary. The compiler won't drop union fields.

Reply:
Don't think this applies anymore

Reply:
The `check_place` call is still unnecessary.

Reply:
Ah, thought you meant checking the locals; I see. That was already there before but I see what you mean where it won't drop union fields (as that's the only place it'll return an error).